src/heartbeat/controllers/heartbeatController.ts

```diff
@@ -45,4 +53,16 @@ export class HeartbeatController {
       next(err);
     }
   };
+
+  public getHeartbeatByTaskId: GetHeartbeatHandler = async (req, res, next) => {
+    try {
+      const record = await this.manager.GetHeartbeatById(req.params);
+      if (record === null) {
```

Comment:
move this condition to the 'heartbeatManager.ts' file inside GetHeartBeatById function to keep controller clean as we can

Reply:
![image](https://github.com/user-attachments/assets/5dc49a15-1873-47e8-b0ef-413295c4286d)

Reply:
![image](https://github.com/user-attachments/assets/7f8e8c3f-f0c2-41a8-a41c-9343bbac4b25)

Reply:
and add req.id to the info log